### ISSUE TITLE
Pywinauto has issues with single file executable under Python 3.9

### ISSUE THREAD

**Issue by Zylanx** [pyinstaller/pyinstaller#5327]
## Description of the issue
Under Python 3.9, when freezing a module that imports pywinauto
win32ui will say that it is unable to find Python39.dll

It works find with single directory and also works fine under Python 3.8 with both single file and single folder.

## Context information (for bug reports)

* Output of `pyinstaller --version`:  
```
4.0, 4.1.dev0
```
* Version of Python: e.g. 3.7
```
Python 3.9
```
* Platform: e.g GNU/Linux (distribution), Windows (language settings), OS X, conda, FreeBSD
```
Windows 10
```
* Did you also try this on another platform? Does it work there?
```
No, it is windows only.
```
* try the latest development version, using the following command:

```shell
pip install https://github.com/pyinstaller/pyinstaller/archive/develop.zip
```

* follow *all* the instructions in our "If Things Go Wrong" Guide
  (https://github.com/pyinstaller/pyinstaller/wiki/If-Things-Go-Wrong) and

### Make sure [everything is packaged correctly](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs#make-sure-everything-is-packaged-correctly)

  * [x] start with clean installation
  * [x] use the latest development version
  * [x] Run your frozen program **from a command window (shell)** — instead of double-clicking on it
  * [x] Package your program in **--onedir mode**
  * [x] Package **without UPX**, say: use the option `--noupx` or set `upx=False` in your .spec-file
  * [x] Repackage you application in **verbose/debug mode**. For this, pass the option `--debug` to `pyi-makespec` or `pyinstaller` or use `EXE(..., debug=1, ...)` in your .spec file.

### A minimal example program which shows the error

```
import pywinauto
```

### Stacktrace / full error message

```
Traceback (most recent call last):
  File "test.py", line 1, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "C:\Users\mattz\AppData\Roaming\Python\Python39\site-packages\PyInstaller\loader\pyimod03_importers.py", line 493, in exec_module
    exec(bytecode, module.__dict__)
  File "pywinauto\__init__.py", line 89, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "C:\Users\mattz\AppData\Roaming\Python\Python39\site-packages\PyInstaller\loader\pyimod03_importers.py", line 493, in exec_module
    exec(bytecode, module.__dict__)
  File "pywinauto\findwindows.py", line 42, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "C:\Users\mattz\AppData\Roaming\Python\Python39\site-packages\PyInstaller\loader\pyimod03_importers.py", line 493, in exec_module
    exec(bytecode, module.__dict__)
  File "pywinauto\controls\__init__.py", line 36, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "C:\Users\mattz\AppData\Roaming\Python\Python39\site-packages\PyInstaller\loader\pyimod03_importers.py", line 493, in exec_module
    exec(bytecode, module.__dict__)
  File "pywinauto\controls\uiawrapper.py", line 42, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "C:\Users\mattz\AppData\Roaming\Python\Python39\site-packages\PyInstaller\loader\pyimod03_importers.py", line 493, in exec_module
    exec(bytecode, module.__dict__)
  File "pywinauto\backend.py", line 35, in <module>
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "C:\Users\mattz\AppData\Roaming\Python\Python39\site-packages\PyInstaller\loader\pyimod03_importers.py", line 493, in exec_module
    exec(bytecode, module.__dict__)
  File "pywinauto\base_wrapper.py", line 44, in <module>
RuntimeError: The win32ui module could not initialize the application object.
[26560] Failed to execute script test
```

**Comment by rokm** [comment]
The latest `pywin32` release (v.`300`) is incompatible with PyInstaller's `onefile` mode - see pywin32/pywin32#1614. 

So you'll have to downgrade `pywin32` to v.`228` (which is probably what is installed in your python 3.8 environment as well).

**Comment by Zylanx** [comment]
I was using pywin32==228.
It seems like an issue that has started happening again with Python 3.9 for some reason?

**Comment by rokm** [comment]
Hmm, you're right, but you're probably getting different errors with `228` and `300`.

`pywin32` `228` doesn't seem to be compatible with python 3.9 - the following exception is raised and displayed in console, for both unfrozen and frozen program:
```
    import win32ui
ImportError: DLL load failed while importing win32ui: A dynamic link library (DLL) initialization routine failed.
```

On the other hand, `300` is incompatible with PyInstaller's `onefile` mode regardless of python version, and it gives the error dialog with
```
The application can not locate Python39.dll (126)
The specified module could not be found.
```
followed by the following exception displayed in the console:
```
    import win32ui
RuntimeError: The win32ui module could not initialize the application object.
```

**Comment by Zylanx** [comment]
Now that you mention it, yes. The errors are different.
I didn't notice that before. I tried both 300 and 228 and just didn't look as hard when I knew that both did not work.

Do you think that means this is a further issue with pywin32 that means that Python 3.9 is just not viable when pywin32 is needed?

**Comment by rokm** [comment]
> Do you think that means this is a further issue with pywin32 that means that Python 3.9 is just not viable when pywin32 is needed?

At the moment, that is indeed the case. From what I can tell, the attempt to fix `pywin32` `228` under python 3.9 accidentally broke compatiblity with PyInstaller's `onefile` builds for all python versions.

So if you want to use `pywin32`, you'll need to switch back to python 3.8.

Alternatively, you can try patching and rebuilding `pywin32`. According to my (rather cursory) tests, [this block](https://github.com/mhammond/pywin32/blob/85e7b248c39a51a2b495afc028b13ecee05f91c3/Pythonwin/Win32uiHostGlue.h#L156-L162) needs to be skipped if python DLL is already loaded, i.e.:
```c
...
    hModCore = GetModuleHandle(py_dll); // Check if Python is already loaded
    if (hModCore == NULL) {
        const int ncandidates = sizeof(py_dll_candidates) / sizeof(py_dll_candidates[0]);
        for (int i = 0; i < ncandidates && hModCore == NULL; i++) {
            wsprintf(fname, _T("%s\\%s\\%s"), app_dir, py_dll_candidates[i], py_dll);
            hModCore = LoadLibrary(fname);
        }
    }
...
```
